packages/node/src/executors/execute/execute.impl.ts

```diff
@@ -57,68 +71,66 @@ export async function* executeExecutor(
     }
   }
 
-  for await (const event of startBuild(options, context)) {
-    if (!event.success) {
-      logger.error('There was an error with the build. See above.');
-      logger.info(`${event.outfile} was not restarted.`);
+  for await (const event of scheduleBuildAndProcessEvents(
+    startBuild(options, context),
+    processRunner
+  )) {
+    if (event.type === 'build') {
+      if (!event.value.success) {
+        logger.error('There was an error with the build. See above.');
+        logger.info(`${event.value.outfile} was not restarted.`);
+      }
     }
-    await handleBuildEvent(event, options);
-    yield event;
-  }
-}
 
-function runProcess(event: NodeBuildEvent, options: NodeExecuteBuilderOptions) {
-  if (subProcess || !event.success) {
-    return;
+    if (event.type === 'build' || options.emitSubprocessEvents) {
+      yield event.value;
+    }
   }
-
-  subProcess = fork(event.outfile, options.args, {
-    execArgv: getExecArgv(options),
-  });
 }
 
-function getExecArgv(options: NodeExecuteBuilderOptions) {
-  const args = ['-r', 'source-map-support/register', ...options.runtimeArgs];
-
-  if (options.inspect === true) {
-    options.inspect = InspectType.Inspect;
-  }
-
-  if (options.inspect) {
-    args.push(`--${options.inspect}=${options.host}:${options.port}`);
-  }
-
-  return args;
-}
+const nextIterValue = (typedIter: TypedIterator) =>
```

Comment:
Typing this is quite challenging. We need to indicate that the result of the iterator is either one type or the other so that we can map the actual value and decorate the result with some data needed to "schedule" them.

Reply:
Am I reading it correctly that both branches of the ternary here are identical?

Reply:
Fixed